gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java

```diff
@@ -215,7 +215,9 @@ protected void executeCancellation() {
     if (this.jobSubmitted) {
       try {
         log.info("[DELETE] workflow {}", this.helixWorkFlowName);
-        this.helixTaskDriver.delete(this.helixWorkFlowName);
+        if (this.cancellationRequested) {
```

Comment:
Is there a cancellation/fail call? The delete here will mean for normal execution the workflow is retained, but for delete it is immediately removed.

Reply:
No, currently there is no cancellation/fail call.
I have created a ticket with helix team HELIX-1180, but I think it will take some time.